src/consent.ts

```diff
@@ -0,0 +1,496 @@
+import {
+    CCPAConsentState,
+    ConsentState,
+    GDPRConsentState,
+    PrivacyConsentState,
+} from '@mparticle/web-sdk';
+import { MParticleUser, MParticleWebSDK } from './sdkRuntimeModels';
+import { isObject } from './utils';
+
+interface GDPRConsentObject {
+    purpose: string;
+}
+
+interface CCPAConsentObject {
+    purpose: string;
+}
+
+interface ConsentJSONObject {
+    gdpr?: Partial<GDPRConsentObject>;
+    ccpa?: Partial<CCPAConsentObject>;
+}
+
+export interface ConsentRuleValues {
+    consentPurpose: string;
+    hasConsented: boolean;
+}
+export interface ConsentRules {
+    includeOnMatch: boolean;
+    values: ConsentRuleValues[];
+}
+
+export interface IConsentSerialization {
+    toMinifiedJsonObject: (state: ConsentState) => ConsentJSONObject;
+    fromMinifiedJsonObject: (json: ConsentJSONObject) => ConsentState;
+}
+
+export interface IConsent {
+    mPinstance;
+    isEnabledForUserConsent: (consentRules: any, user: any) => boolean;
+    createPrivacyConsent: (
+        consented: boolean,
```

Comment:
we should extract these 5 properties into an interface, its reused in several other places, ~lines 160 and 360 in this file [consent.ts] for example